pkg/expression/contextstatic/evalctx.go

```diff
@@ -203,6 +203,13 @@ func WithParamList(params *variable.PlanCacheParamList) StaticEvalCtxOption {
 	}
 }
 
+// WithEnableRedactLog sets the value of the 'tidb_redact_log' system variable.
+func WithEnableRedactLog(enableRedactLog string) StaticEvalCtxOption {
```

Comment:
This PR actually finds one bug :beers: . The `enableRedactLog` is not copied to the detached expr context.